pragma-node/src/infra/repositories/onchain_repository.rs

```diff
@@ -380,40 +380,42 @@ async fn get_publisher_with_components(
 ) -> Result<Publisher, InfraError> {
     let raw_sql_entries = format!(
         r#"
-        WITH recent_entries AS (
-            SELECT 
-                pair_id,
-                price,
-                source,
-                timestamp AS last_updated_timestamp
-            FROM 
-                {table_name}
-            WHERE
-                publisher = '{publisher_name}'
-                AND timestamp >= NOW() - INTERVAL '1 day'
-        ),
-        ranked_entries AS (
-            SELECT 
-                pair_id,
-                price,
-                source,
-                last_updated_timestamp,
-                ROW_NUMBER() OVER (PARTITION BY pair_id, source ORDER BY last_updated_timestamp DESC) as rn
-            FROM 
-                recent_entries
-        )
+    WITH recent_entries AS (
         SELECT 
             pair_id,
             price,
             source,
-            last_updated_timestamp
+            timestamp AS last_updated_timestamp
         FROM 
-            ranked_entries
-        WHERE 
-            rn = 1
-        ORDER BY 
-            pair_id, source ASC;
-        "#,
+            {table_name}
+        WHERE
+            publisher = '{publisher_name}'
+            AND timestamp >= NOW() - INTERVAL '1 day'
+    ),
+    ranked_entries AS (
+        SELECT 
+            pair_id,
+            price,
+            source,
+            last_updated_timestamp,
+            ROW_NUMBER() OVER (PARTITION BY pair_id, source ORDER BY last_updated_timestamp DESC) as rn,
+            COUNT(*) OVER (PARTITION BY pair_id, source) as daily_updates
```

Comment:
We also need to add `daily_updates` in the `RawLastPublisherEntryForPair` & `PublisherEntry` structs right?

Currently I think it will get dropped when we convert the sql result to PublisherEntry

Reply:
yes sorry wasnt ready for review